src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_api.py

```diff
@@ -554,3 +555,8 @@ def custom_dont_trace_external_files(abs_path):
         if reset_caches:
             py_db.clear_dont_trace_start_end_patterns_caches()
 
+    def stop_on_entry(self):
+        main_thread = pydevd_utils.get_main_thread()
```

Comment:
I think it'd be nice to check if `main_thread` is not `None` here (and if it is show a `pydev_log.critical()` to the user) just to be on the safe side.